test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -311,12 +311,35 @@ public void Single_unpooled_connection()
     }
 
     [Test, IssueLink("https://github.com/npgsql/npgsql/issues/4963")]
-    public void Single_unpooled_closed_connection()
+    public void Single_closed_connection_in_transaction_scope([Values] bool pooling)
     {
         using var dataSource = CreateDataSource(csb =>
         {
-            csb.Pooling = false;
+            csb.Pooling = pooling;
+            csb.Enlist = true;
+        });
+
+        using (var scope = new TransactionScope())
+        using (var conn = dataSource.OpenConnection())
+        using (var cmd = new NpgsqlCommand("SELECT 1", conn))
+        {
+            cmd.ExecuteNonQuery();
+            conn.Close();
+            Assert.That(pooling ? dataSource.Statistics.Busy : dataSource.Statistics.Total, Is.EqualTo(1));
+            scope.Complete();
+        }
+
+        Assert.That(pooling ? dataSource.Statistics.Busy : dataSource.Statistics.Total, Is.EqualTo(0));
+    }
+
+    [Test, IssueLink("https://github.com/npgsql/npgsql/issues/5783")]
+    public void Single_closed_connection_in_transaction_scope_multiple_hosts([Values] bool pooling)
```

Comment:
nit: can merge into the above test, using a second `[Values]` parameter for multihost